src/main/java/org/sagebionetworks/web/client/widget/team/JoinTeamWidget.java

```diff
@@ -365,10 +425,14 @@ public void sendJoinRequestStep3() {
 		synapseClient.requestMembership(authenticationController.getCurrentUserPrincipalId(), teamId, message, gwt.getHostPageBaseURL(), new AsyncCallback<Void>() {
 			@Override
 			public void onSuccess(Void result) {
-				String successJoinMessage = successMessage == null ? WidgetConstants.JOIN_TEAM_DEFAULT_SUCCESS_MESSAGE : successMessage;
-				String message = isAcceptingInvite ? successJoinMessage : "Request Sent";
-				view.showInfo(message, "");
-				refresh();
+				if (teamMembershipStatus.getCanJoin()) {
+					refresh();
+				} else {
+					view.showInfo("Request Sent", "");
```

Comment:
Verify if user creates a new join team request, that both the "Request Sent" message is popped up, and the Join button UI is right (shows a request pending approval message).